hwmv2: arm: intel_socfpga_std: Port to HWMv2

golowanow opened the pull request:
Move and convert to HWMv2 `soc/arm/intel_socfpga_std` SoC family `INTEL_SOCFPGA_STD` with its SoC series `CYCLONE5` and board configuration `cyclonev_socdk`

golowanow committed:
soc: arm: intel_socfpga_std: Move and convert to HWMv2

Move and convert to HWMv2 `soc/arm/intel_socfpga_std`
SoC family configuration, its SoC series `cyclonev`,
and SoC `intel_socfpga_std_cyclonev`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm: cyclonev_socdk: Move to HWMv2

Move and convert to HWMv2 `cyclonev_socdk` board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented on boards/intel/cyclonev_socdk/Kconfig.defconfig:
```diff
@@ -1,14 +1,8 @@
-
-config BOARD
-	default "cyclonev_socdk"
-depends on BOARD_CVSXDEVKIT
```
`BOARD_CVSXDEVKIT` becomes `BOARD_CYCLONEV_SOCDK` with HWMv2

golowanow commented:
FYI @mevalver (#45133)

nordicjm commented on soc/intel/intel_socfpga_std/cyclonev/Kconfig:
```diff
@@ -1,14 +1,14 @@
-	help
-		Support for Intel SoC FPGA Series
+
+config SOC_CYCLONE5
+	select SOC_SERIES_CYCLONE5
```
Goes in Kconfig.soc

golowanow replied:
changed

nordicjm commented on soc/intel/intel_socfpga_std/cyclonev/Kconfig.soc:
```diff
@@ -0,0 +1,20 @@
+	  "Intel SoC FPGA Cyclone5"
+
+config SOC
+	default "intel_socfpga_std_cyclonev" if SOC_CYCLONE5
```
```suggestion
	default "cyclone5" if SOC_CYCLONE5
```

golowanow replied:
why to rename as `cyclone5` ?
what is wrong with `intel_socfpga_std_cyclonev`, or would it be `cyclonev` ?

nordicjm replied:
`	default "cyclonev" if SOC_CYCLONEV` can do if 5 is updated to V

golowanow replied:
@nordicjm,  now I'm lost (( can you please point me where it is set as mandatory that all these names must match exactly to each other for v.2 ?

nordicjm replied:
Correct me if I'm wrong but @nashif was complaining about the length of board names and `cyclonev_socdk/cyclonev` seems a lot easier to type than `cyclonev_socdk/intel_socfpga_std_cyclonev`

golowanow replied:
> Correct me if I'm wrong but @nashif was complaining about the length of board names and `cyclonev_socdk/cyclonev` seems a lot easier to type than `cyclonev_socdk/intel_socfpga_std_cyclonev`

This board has no other SoC or variants and its SoC name seems to follow naming convention to include vendor and SoC family/series.
Currently this PR generates almost the same .config file except `BOARD_CVSXDEVKIT` which looks like a leftover and conflicts with the v.2 scheme board name. 

>  .. can do if 5 is updated to V

From what I see from the SoC description `V` is a part of its official name, although `cyclone5` naming is also used in .dts, 
m.b. @mevalver can explain why there is this difference.

In my opinion adhoc renaming or optimizing should be minimal for this HWMv2 migration effort unless some vendor decides to apply its thoroughly developed naming scheme to align all its SoC/boards in the process.

mevalver replied:
@golowanow no difference, its just a naming thing, you can called cyclone5 or cyclonev

golowanow replied:
> @golowanow no difference, its just a naming thing, you can called cyclone5 or cyclonev

@mevalver, there is similar and related renaming question in regard of `./arm64/intel_socfpga_agilex_socdk` boards
https://github.com/zephyrproject-rtos/zephyr/pull/68909#issuecomment-1948592044

In case of `intel_socfpga_std` there are at least 2 options:

1) just keep naming 'as-is' and what  this PR does at the moment:
SoC - `intel_socfpga_std_cyclonev`, board - `cyclonev_socdk`
but using HWMv2 full `board/soc` naming it looks redundant: `cyclonev_socdk/intel_socfpga_std_cyclonev`

2) rename the SoC/board to some short form and in alignment with what might be chosen for `intel_socfpga`:
`intel_socfpga_std/cyclonev_socdk`
or:
`intel_socfpga_std_socdk/cyclonev`
or:
`intel_socfpga_std/cyclonev`

nordicjm replied:
name and text should match so e.g. `default "cyclone5" if SOC_CYCLONE5` or `default "cyclonev" if SOC_CYCLONEV`

nordicjm commented on soc/intel/intel_socfpga_std/soc.yml:
```diff
@@ -0,0 +1,6 @@
+    series:
+      - name: cyclonev
+        socs:
+          - name: intel_socfpga_std_cyclonev
```
```suggestion
          - name: cyclone5
```

golowanow replied:
changed

nordicjm commented on boards/intel/cyclonev_socdk/Kconfig.cyclonev_socdk:
```diff
@@ -0,0 +1,10 @@
+# Cyclone V SoC development kit configuration option
+
+config BOARD_CYCLONEV_SOCDK
+	bool
```
```suggestion
```

golowanow replied:
removed

nordicjm commented on boards/intel/cyclonev_socdk/board.yml:
```diff
@@ -0,0 +1,4 @@
+board:
+  name: cyclonev_socdk
+  socs:
+    - name: intel_socfpga_std_cyclonev
```
```suggestion
    - name: cyclone5
```

golowanow replied:
changed

golowanow commented on soc/intel/intel_socfpga_std/cyclonev/Kconfig:
```diff
@@ -1,14 +1,14 @@
-	bool "Intel SoC FPGA Cyclone5 Series"
 	select ARM
 	select CPU_CORTEX_A9
 	select SOC_FAMILY_INTEL_SOCFPGA_STD
```
changed

gmarull reviewed (dismissed):
update docs/maintainers as needed

golowanow committed:
soc: arm: intel_socfpga_std: Align board subdirectory

Aligh board directory to other Intel FPGA boards moving
`boards/intel/cyclonev_socdk` -->
`boards/intel/intel_socfpga_std/cyclonev_socdk`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
* rebased to the latest changes in `collab-hwm`
* board directory aligned to #68909
 `boards/intel/cyclonev_socdk` --> `boards/intel/intel_socfpga_std/cyclonev_socdk`

golowanow commented:
@nashif, there is no explicit record in the `MAINTAINERS` for this SoC/Board, should we add some ?

nordicjm commented on soc/intel/intel_socfpga_std/Kconfig:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2022-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+# Description:
+# Adding support to Cyclone V SoC FPGA
```
not really sure why there is what looks to be a changelog here and in the other files?

nordicjm commented on soc/intel/intel_socfpga_std/Kconfig.defconfig:
```diff
@@ -0,0 +1,6 @@
+# Description:
+# Adding support for Cyclone V SoC FPGA
+
+rsource "*/Kconfig.defconfig.series"
```
```suggestion
if SOC_FAMILY_INTEL_SOCFPGA_STD

rsource "*/Kconfig.defconfig.series"

endif # SOC_FAMILY_INTEL_SOCFPGA_STD
```

golowanow replied:
changed

nordicjm commented on soc/intel/intel_socfpga_std/Kconfig:
```diff
@@ -0,0 +1,6 @@
+# Description:
+# Adding support to Cyclone V SoC FPGA
+
+rsource "*/Kconfig"
```
```suggestion
if SOC_FAMILY_INTEL_SOCFPGA_STD

rsource "*/Kconfig"

endif # SOC_FAMILY_INTEL_SOCFPGA_STD
```

golowanow replied:
changed

nordicjm commented on boards/intel/cyclonev_socdk/Kconfig.cyclonev_socdk:
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+# Description:
+# Cyclone V SoC development kit configuration option
```
let's remove this comment from every file

golowanow committed:
soc: board: intel_socfpga_std: Align names to 'Cyclone V'

Align all names to `cyclonev` instead of using `cyclone5`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: arm: intel_socfpga_std: Rename with HWMv2

Rename SoC for better HWMv2 use:
 `intel_socfpga_std_cyclonev` --> `cyclonev`

The resulting shortened board name:
 `cyclonev_socdk/intel_socfpga_std_cyclonev` -->
 `cyclonev_socdk/cyclonev`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
* Align all names to `cyclonev` (instead of occasional `cyclone5`).

* Rename SoC for HWMv2 use:  `intel_socfpga_std_cyclonev` --> `cyclonev`
the full board name:  `cyclonev_socdk/intel_socfpga_std_cyclonev` -->  `cyclonev_socdk/cyclonev`
